pkg/apis/pipeline/v1/pipeline_validation_test.go

```diff
@@ -3434,11 +3435,30 @@ func TestValidateParamArrayIndex_valid(t *testing.T) {
 			}},
 		},
 		params: []Param{{Name: "second-param", Value: *NewStructuredValues("second-value", "second-value-again")}},
+	}, {
+		name: "validation with alpha gate",
+		original: PipelineSpec{
+			Params: []ParamSpec{
+				{Name: "first-param", Type: ParamTypeArray, Default: NewStructuredValues("default-value", "default-value-again")},
+				{Name: "second-param", Type: ParamTypeString},
+			},
+			Tasks: []PipelineTask{{
+				Params: []Param{
+					{Name: "first-task-first-param", Value: *NewStructuredValues("$(params.first-param[1])")},
+					{Name: "first-task-second-param", Value: *NewStructuredValues("$(params.second-param[0])")},
+					{Name: "first-task-third-param", Value: *NewStructuredValues("static value")},
+				},
+			}},
+		},
+		params:    []Param{{Name: "second-param", Value: *NewStructuredValues("second-value", "second-value-again")}},
+		apiFields: config.AlphaAPIFields,
```

Comment:
can you also add a test for beta api fields?

Reply:
beta api fields is a default value in this test function, that's why I didn't add a specific case for that. Do you think we should add that one?

Reply:
Hm I'm not really seeing where the default is set to beta, can you point me to it?
I think it would be better to make it explicit-- what if someone changes the default?